packages/cadl-ranch-specs/http/payload/json-merge-patch/mockapi.ts

```diff
@@ -0,0 +1,84 @@
+import { passOnSuccess, ScenarioMockApi, mockapi, json, MockApi } from "@azure-tools/cadl-ranch-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+function createMockApis(route: string, isUpdateRequest: boolean): MockApi {
+  const url = `/json-merge-patch/${route}`;
+  const expectedCreateBody = {
+    name: "Madge",
+    description: "desc",
+    map: {
+      key: {
+        name: "InnerMadge",
+        description: "innerDesc",
+      },
+    },
+    array: [
+      {
+        name: "InnerMadge",
+        description: "innerDesc",
+      },
+    ],
+    intValue: 1,
+    floatValue: 1.1,
+    innerModel: {
+      name: "InnerMadge",
+      description: "innerDesc",
+    },
+    intArray: [1, 2, 3],
+  };
+  const expectedUpdateBody = {
+    name: "Madge",
+    description: null,
+    map: {
+      key: {
+        name: "InnerMadge",
+        description: null,
+      },
+    },
+    array: null,
+    intValue: null,
+    floatValue: null,
+    innerModel: null,
+    intArray: null,
+  };
+  if (isUpdateRequest) {
+    return mockapi.patch(url, (req) => {
+      req.expect.deepEqual(req.body.description, expectedUpdateBody.description);
+      req.expect.deepEqual(req.body.map, expectedUpdateBody.map);
+      req.expect.deepEqual(req.body.array, expectedUpdateBody.array);
+      req.expect.deepEqual(req.body.intValue, expectedUpdateBody.intValue);
+      req.expect.deepEqual(req.body.floatValue, expectedUpdateBody.floatValue);
+      req.expect.deepEqual(req.body.innerModel, expectedUpdateBody.innerModel);
```

Comment:
Same here, verification should only about the `description`, not `name` which is not modified.